xrpl-wasm-stdlib/src/core/types/vector_256.rs

```diff
@@ -0,0 +1,291 @@
+//!
+//! This module provides support for XRPL's Vector256 field type, which represents
+//! an array of 32-byte (256-bit) hashes. Common uses include CredentialIDs, NFTokenOffers,
+//! and other fields that contain multiple hash values.
+//!
+//! ## Serialization Format
+//!
+//! Vector256 fields are serialized as:
+//! - 1 byte: total byte length of the data (must be a multiple of 32)
+//! - N × 32 bytes: the hash values
+//!
+//! The count of hashes is calculated as: data_length / 32
+//!
+//! ## Example
+//!
+//! ```no_run
+//! use xrpl_wasm_stdlib::core::types::vector_256::Vector256;
+//! use xrpl_wasm_stdlib::core::types::uint::Hash256;
+//!
+//! // Parse from raw bytes (64 = 2 hashes × 32 bytes each)
+//! let raw_data = [64, /* 64 bytes of hash data */];
+//! let vector = Vector256::from_bytes(&raw_data).unwrap();
+//! assert_eq!(vector.len(), 2);
+//! ```
+
+use crate::core::types::uint::Hash256;
+use crate::host::Error;
+
+/// Maximum number of hashes that can be stored in a Vector256
+pub const MAX_VECTOR_256_SIZE: usize = 8;
+
+/// A collection of 256-bit hashes with a maximum size
+#[derive(Debug, Clone, Eq, PartialEq)]
+#[repr(C)]
+pub struct Vector256 {
+    /// The hash values
+    pub hashes: [Hash256; MAX_VECTOR_256_SIZE],
+    /// Number of valid hashes (0 to MAX_VECTOR256_SIZE)
+    pub len: u8,
+}
+
+impl Vector256 {
+    /// Creates an empty Vector256
+    pub const fn new() -> Self {
+        use crate::core::types::uint::UInt;
+        const EMPTY_HASH: Hash256 = UInt([0u8; 32]);
+        Self {
+            hashes: [EMPTY_HASH; MAX_VECTOR_256_SIZE],
+            len: 0,
+        }
+    }
+
+    /// Returns the number of hashes in the vector
+    pub fn len(&self) -> usize {
+        self.len as usize
+    }
+
+    /// Returns true if the vector is empty
+    pub fn is_empty(&self) -> bool {
+        self.len == 0
+    }
+
+    /// Returns a slice of the valid hashes
+    pub fn as_slice(&self) -> &[Hash256] {
+        &self.hashes[..self.len as usize]
+    }
+
+    /// Gets a hash at the specified index, if it exists
+    pub fn get(&self, index: usize) -> Option<&Hash256> {
+        if index < self.len as usize {
+            Some(&self.hashes[index])
+        } else {
+            None
+        }
+    }
+
+    /// Parses a Vector256 from raw bytes
+    ///
+    /// The byte array should be in the format:
+    /// - First byte: total byte length of the data (not count of hashes)
+    /// - Remaining bytes: 32-byte hashes
+    ///
+    /// Returns an error if:
+    /// - The buffer is too small
+    /// - The data length is not a multiple of 32
+    /// - The count exceeds MAX_VECTOR256_SIZE
+    /// - The buffer doesn't contain enough bytes for all hashes
+    pub fn from_bytes(bytes: &[u8]) -> crate::host::Result<Self> {
+        use crate::host::trace::{trace, trace_num};
+
+        let _ = trace("Vector256::from_bytes called");
+        let _ = trace_num("  bytes.len():", bytes.len() as i64);
```

Comment:
There shouldn't be any traces in the successful path. And minimal traces in the error paths.

Reply:
Totally agreed, that was a remnant from me trying to debug something. Nice catch!

Fixed via ec7a423da593a54859313df67776c3cc297afa77